src/Symfony/Component/Cache/Traits/Item.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Symfony\Contracts\Cache\ItemInterface;
+
+/**
+ * A short namespace-less class to serialize items with metadata.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+class Ͼ
+{
+    public const METADATA_EXPIRY_OFFSET = 1527506807;
+
+    public readonly mixed $value;
+    public readonly array $metadata;
+
+    public function __construct(mixed $value, array $metadata)
+    {
+        $this->value = $value;
+        $this->metadata = $metadata;
+    }
+
+    public function __serialize(): array
+    {
+        $pack = pack('VN', (int) (0.1 + ($this->metadata['expiry'] ?: 3674990454) - self::METADATA_EXPIRY_OFFSET), $this->metadata['ctime'] ?? 0);
```

Comment:
@nicolas-grekas  3674990454 is greater than PHP_INT_MAX on 32-bit platforms
Why not to use 
```
$length = 4 + ($ctime <= 255 ? 1 : ($ctime <= 65535 ? 2 : ($ctime <= 16777215 ? 3 : 4)));
$value['^'] = substr(pack('NV', (int) ceil($metadata[CacheItem::METADATA_EXPIRY]), $ctime), 0, $length);
```
without magic numbers?

Reply:
>  3674990454 is greater than PHP_INT_MAX on 32-bit platforms

yes, but the computation still works as expected when using floats, to that's fine on 32b too

Reply:
> Why not to use [variable length packing]

I'm not super convinced by the current way. Maybe we can implement our variant of this instead, but revisited to handle 32bits numbers instead of 64b:
https://sqlite.org/src4/doc/trunk/www/varint.wiki

Reply:
ctime is now packed as a 14-bits floating point number. This leave 2 bits free: one is use to tell if tags are set or not, the other could be used for future needs.